hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/authentication/AuthorizationStatus.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs.authentication;
+
+import org.apache.hadoop.fs.azurebfs.contracts.exceptions.AbfsAuthorizationException;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResource;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResourceResult;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResult;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations.DEFAULT_SAS_REFRESH_INTERVAL_BEFORE_EXPIRY;
+
+/**
+ * AuthorizationStatus maintains the status of Authorization and also SAS
+ * token if authorizer is providing it.
+ */
+public class AuthorizationStatus {
+  private boolean isAuthorized;
+  private HashMap<URI, SasTokenData> sasTokenMap;
```

Comment:
Declare type as `Map<URI, SasTokenData>` and mark final. 

What is the synchronization policy here if a set of sas tokens are being shared. Should it be some concurrent hash map?

Reply:
AuthorizationStatus instance is saved at a time by the Abfs inpt or output stream, which will be operating one I/O operation at a time.
As the instance is not shared at Abfs filesystem level across requests, concurrency doesnt need handling,

Reply:
Code is modified. Resolving this comment.